python/ml4ir/applications/ranking/tests/test_metrics.py

```diff
@@ -88,6 +151,35 @@ def test_model_training(self):
             assert gold_metric_name in metrics
             assert np.isclose(metrics[gold_metric_name], gold_metric_val, atol=0.05)
 
+    def test_stat_sig_evaluation(self):
+        """testing ml4ir stat sig computation end-to-end"""
+
+        ROOT_DATA_DIR = "ml4ir/applications/ranking/tests/data"
+        feature_config_path = os.path.join(
+            ROOT_DATA_DIR, "configs", "feature_config_aux_loss.yaml"
+        )
+        model_config_path = os.path.join(
+            ROOT_DATA_DIR, "configs", "model_config_cyclic_lr.yaml")
+        eval_config_path = os.path.join(
+            ROOT_DATA_DIR, "configs", "aux_metrics_evaluation_config.yaml")
+
+        data_dir = os.path.join(ROOT_DATA_DIR, "tfrecord")
+        aux_loss = "aux_one_hot_cross_entropy"
+        self.train_ml4ir(data_dir, feature_config_path,
+                    model_config_path, eval_config_path, self.log_dir, aux_loss)
```

Comment:
It's an end to end test. thought would be better. It's a small model with small dataset with just 2 epochs.

Reply:
The problem is there is a Keras memory leak everytime we create a model. :/

I'll approve for now, but we should try to avoid end-to-end tests as much as possible.

Reply:
The clean way to do this is to mock the `.predict()` function and just call `RankingModel.evaluate()` without ever creating a Keras model.